mm2src/coins/utxo/rpc_clients.rs

```diff
@@ -1037,53 +1037,43 @@ pub fn spawn_electrum(
     Ok(electrum_connect(req.url.clone(), config, event_handlers))
 }
 
-#[cfg(target_arch = "wasm32")]
-#[wasm_bindgen(raw_module = "../../../js/defined-in-js.js")]
-extern "C" {
-    fn host_electrum_connect(ptr: *const c_char, len: i32) -> i32;
-    fn host_electrum_is_connected(ri: i32) -> i32;
-    fn host_electrum_request(ri: i32, ptr: *const c_char, len: i32) -> i32;
-    fn host_electrum_reply(ri: i32, id: i32, rbuf: *mut c_char, rcap: i32) -> i32;
-}
-
+/// Attempts to process the request (parse url, etc), build up the config and create new electrum connection
 #[cfg(target_arch = "wasm32")]
 pub fn spawn_electrum(
     req: &ElectrumRpcRequest,
-    _event_handlers: Vec<RpcTransportEventHandlerShared>,
+    event_handlers: Vec<RpcTransportEventHandlerShared>,
 ) -> Result<ElectrumConnection, String> {
-    use std::net::{IpAddr, Ipv4Addr};
+    let mut url = req.url.clone();
+    let uri: Uri = try_s!(req.url.parse());
 
-    let args = json::to_vec(req).unwrap();
-    let rc = host_electrum_connect(args.as_ptr() as *const c_char, args.len() as i32);
-    if rc < 0 {
-        panic!("!host_electrum_connect: {}", rc)
+    if uri.scheme().is_some() {
+        return ERR!(
+            "There has not to be a scheme in the url: {}. \
+            'ws://' scheme is used by default. \
+            Consider using 'protocol: \"SSL\"' in the electrum request to switch to the 'wss://' scheme.",
+            url
+        );
     }
-    let ri = rc; // Random ID assigned by the host to connection.
-
-    let responses = Arc::new(AsyncMutex::new(HashMap::new()));
-    let tx = Arc::new(AsyncMutex::new(None));
 
     let config = match req.protocol {
-        ElectrumProtocol::TCP => ElectrumConfig::TCP,
+        ElectrumProtocol::TCP => {
```

Comment:
For clarity, please extend the `ElectrumProtocol` with `WS` and `WSS` variants instead. TCP and SSL should not be available for wasm32 at least in the browser.

Reply:
It makes sense! I followed a path of least resistance. Will do it!

Reply:
Done